Panic introduced in context deadline exceeded errors patch

I had a cluster fail (https://github.com/cockroachdb/cockroach/issues/35178) and wanted to run a debug.zip on the cluster. I previously knew that it was node 5 that died. I ran `Andrews-MBP-2:~ andrewwoods$ roachprod run $CLUSTER:5 -- ./cockroach debug zip --url {pgurl:1} debug.zip` and then hit a panic:

```
E190225 18:00:10.509134 1 util/log/crash_reporting.go:202  a panic has occurred!
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
    panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x24c68db]

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/util/log.RecoverAndReportPanic(0x3914600, 0xc000052090, 0xc0004eb300)
    /go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:100 +0x9f
panic(0x2d763e0, 0x552efd0)
    /usr/local/go/src/runtime/panic.go:513 +0x1b9
github.com/cockroachdb/cockroach/pkg/cli.runDebugZip(0x5558740, 0xc0005f7620, 0x1, 0x3, 0x0, 0x0)
    /go/src/github.com/cockroachdb/cockroach/pkg/cli/zip.go:280 +0xc1b
github.com/cockroachdb/cockroach/pkg/cli.MaybeDecorateGRPCError.func1(0x5558740, 0xc0005f7620, 0x1, 0x3, 0x0, 0x0)
    /go/src/github.com/cockroachdb/cockroach/pkg/cli/error.go:69 +0x6b
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).execute(0x5558740, 0xc0005f75f0, 0x3, 0x3, 0x5558740, 0xc0005f75f0)
    /go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:762 +0x473
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0x5555560, 0x8, 0x0, 0xc000871ed8)
    /go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:852 +0x2fd
github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra.(*Command).Execute(0x5555560, 0x18, 0xc000871ef0)
    /go/src/github.com/cockroachdb/cockroach/vendor/github.com/spf13/cobra/command.go:800 +0x2b
github.com/cockroachdb/cockroach/pkg/cli.Run(0xc00004c1f0, 0x5, 0x5, 0xc000052090, 0xc0004eb300)
    /go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:194 +0x65
github.com/cockroachdb/cockroach/pkg/cli.Main()
    /go/src/github.com/cockroachdb/cockroach/pkg/cli/cli.go:69 +0x222
main.main()
    /go/src/github.com/cockroachdb/cockroach/pkg/cmd/cockroach/main.go:30 +0x20
Error:  exit status 2
```

--- a/pkg/cli/zip.go
+++ b/pkg/cli/zip.go
@@ -271,26 +271,30 @@ func runDebugZip(cmd *cobra.Command, args []string) error {
 					}
 				}
 
-				var stacks *serverpb.JSONResponse
+				var stacksData []byte
 				err = contextutil.RunWithTimeout(baseCtx, "request stacks", timeout,
 					func(ctx context.Context) error {
-						stacks, err = status.Stacks(ctx, &serverpb.StacksRequest{NodeId: id})
+						stacks, err := status.Stacks(ctx, &serverpb.StacksRequest{NodeId: id})
+						if err != nil {
+							stacksData = stacks.Data
+						}
 						return err
 					})
-				if err := z.createRawOrError(prefix+"/stacks", stacks.Data, err); err != nil {
+				if err := z.createRawOrError(prefix+"/stacks", stacksData, err); err != nil {
 					return err
 				}
 
-				var heap *serverpb.JSONResponse
+				var heapData []byte
 				err = contextutil.RunWithTimeout(baseCtx, "request heap profile", timeout,
 					func(ctx context.Context) error {
-						heap, err = status.Profile(ctx, &serverpb.ProfileRequest{
+						heap, err := status.Profile(ctx, &serverpb.ProfileRequest{
 							NodeId: id,
 							Type:   serverpb.ProfileRequest_HEAP,
 						})
+						heapData = heap.Data
 						return err
 					})
-				if err := z.createRawOrError(prefix+"/heap", heap.Data, err); err != nil {
+				if err := z.createRawOrError(prefix+"/heap", heapData, err); err != nil {
 					return err
 				}
 

